domain/blockdag/finality.go

```diff
@@ -18,6 +18,12 @@ func (dag *BlockDAG) FinalityInterval() uint64 {
 	return uint64(dag.Params.FinalityDuration / dag.Params.TargetTimePerBlock)
 }
 
+// pruningDepth is used to determine the depth from the virtual block where the pruning point is set once updated.
+func (dag *BlockDAG) pruningDepth() uint64 {
+	k := uint64(dag.Params.K)
+	return 2*dag.FinalityInterval() + 4*mergeSetSizeLimit*k + 2*k + 2
```

Comment:
Please add unit test that makes sure that this equation evaluates to what we expect it to evaluate in all standard nets.